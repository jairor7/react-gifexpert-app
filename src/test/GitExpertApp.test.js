import { shallow } from "enzyme";
import GitExpertApp from "../GifExpertApp";

describe("Pruebas al componente <GitExpertApp />", () => {
  test("Debe de mostrar el componente correctamente", () => {
    const wrapper = shallow(<GitExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Samurai X", "Dragon Ball"];
    const wrapper = shallow(<GitExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
