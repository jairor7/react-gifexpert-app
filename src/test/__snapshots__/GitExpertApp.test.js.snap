// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GitExpertApp /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={Array []}
    setCategories={[Function]}
  />
  <hr />
  <div />
</Fragment>
`;

exports[`Pruebas al componente <GitExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={
      Array [
        "One Punch",
        "Samurai X",
        "Dragon Ball",
      ]
    }
    setCategories={[Function]}
  />
  <hr />
  <div>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </div>
</Fragment>
`;
